### PR TITLE
Add no-readme option to GitHub tag

### LINKED ISSUE #4029: Add a short version of a Github embed block
**Is your feature request related to a problem? Please describe.**

Currently, embedding of Github repository takes very much page height. I'm usually importing a repository with some context so user don't need to view readme. It's especially actual when you import several repos in a row. Even short article became too long to scroll.

**Describe the solution you'd like**

I'd like to have a solution which allows to hide readme in a liquid tag. It could be a flag `readme`: 
```
{% github author/some-repo readme=no %}
```

**Describe alternatives you've considered**

Alternative solution is to make readme block collapsible. And add an option to manage default state collapsed/uncollapsed.

**Additional context**

What it is now:
Screen Shot 2019-09-15 at 08 37 29

What I'd expect it to be alike:
Screen Shot 2019-09-15 at 08 38 24

### PR DESCRIPTION
**Pull request by johncurcio**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description

Add option ``no-readme`` to GitHub liquid tag that hides the readme file from the preview.

## Related Tickets & Documents

Closes #4029 

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

**Comment by benhalpern** [comment]
I love this idea and approve of it going in. Would love some secondary eyes on parsing, etc. @maestromac @Zhao-Andy

File: app/liquid_tags/github_tag/github_readme_tag.rb [hunk 1]
```diff
@@ -36,8 +38,31 @@ def get_content(link)
...
+    def valid_option(option)
+      option.match(/no-readme/)
+    end
+
+    def parse_options(link)
+      opts = sanitize_link(link)
+      _, *options = opts.split(" ")
+
+      validated_options = options.map { |option| valid_option(option) }.reject(&:nil?)
+      raise StandardError, "Invalid Options" unless options.empty? || !validated_options.empty?
```
Comment (maestromac) [inline app/liquid_tags/github_tag/github_readme_tag.rb]: I think using `any?` here would be more semantic
```suggestion
      raise StandardError, "Invalid Options" unless options.empty? || validated_options.any?
```

File: app/liquid_tags/github_tag/github_readme_tag.rb [hunk 2]
```diff
@@ -36,8 +38,31 @@ def get_content(link)
...
+    def valid_option(option)
+      option.match(/no-readme/)
+    end
+
+    def parse_options(link)
+      opts = sanitize_link(link)
+      _, *options = opts.split(" ")
+
+      validated_options = options.map { |option| valid_option(option) }.reject(&:nil?)
+      raise StandardError, "Invalid Options" unless options.empty? || validated_options.any?
```
Comment (Zhao-Andy) [inline app/liquid_tags/github_tag/github_readme_tag.rb]: I'd prefer a more helpful error message. Right now it shows: `Base: Invalid Options` Maybe change it to:
```suggestion
      raise StandardError, raise StandardError, "GitHub tag: `#{link}`: Invalid option - did you mean `no-readme`?" unless options.empty? || validated_options.any?
```

Comment (johncurcio) [reply app/liquid_tags/github_tag/github_readme_tag.rb]: I like your error message. I'm going to commit your suggestion.

**Pull request merged**